cli/determined_cli/trial.py

```diff
@@ -82,25 +82,36 @@ def describe_trial(args: Namespace) -> None:
 
 @authentication_required
 def logs(args: Namespace) -> None:
-    offset, state = 0, None
-
-    def print_logs(limit: Optional[int] = None) -> None:
-        nonlocal offset, state
-        path = "trials/{}/logsv2?offset={}".format(args.trial_id, offset)
-        if limit:
-            path = "{}&limit=?".format(limit)
-        for log in api.get(args.master, path).json():
+    last_offset, last_state = 0, None
+
+    def print_logs(offset: Optional[int], limit: Optional[int] = 5000) -> Any:
```

Comment:
non-blocking, but let's talk about it later: Having the arguments to a function be the only thing that determines its behavior is, of course, a good thing in general. However, in this case, (1) the remaining use of `nonlocal` means that ideal is somewhat broken anyway, (2) the scope of the function and the things that can affect it is quite small, (3) making use of the variable directly would considerably simplify downstream code -- no other code has to deal with `offset` directly in that case, as we had before, and (4) the argument is usually (but not always) the same as the referenced nonlocal variable anyway, so now there's a confusing (and inconsistently present) duplication.

If it makes you feel better, you can think of it as syntactic sugar for a class containing `offset` and `state` attributes and a `print_logs` method.